apps/yggdrasil/lib/yggdrasil/user.ex

```diff
@@ -1,14 +1,21 @@
 defmodule Yggdrasil.User do
   use Ecto.Schema
   import Ecto.Changeset
+  import Ecto.Query, only: [from: 1, from: 2]
   import Comeonin.Bcrypt, only: [hashpwsalt: 1]
-  require Logger
+
+  alias Yggdrasil.{Repo, User, Role, UserRole}
+
+  require IEx
+
+  @default_role "player"
 
   schema "users" do
     field :username, :string
     field :hash, :string
     field :password, :string, virtual: true # not part of table
     field :password_confirmation, :string, virtual: true # not part of table
+    field :permissions, :any, default: [], virtual: true # :any skips type checking
```

Comment:
`:any` in this case skips type checking against the db types. Being that permissions is a map, I don't think one there is a type for it (could be wrong) and it's not something the db adapters or ecto really needs to enforce.